mender-update/CMakeLists.txt

```diff
@@ -22,18 +22,23 @@ gtest_discover_tests(context_test)
 add_dependencies(check context_test)
 
 
-add_library(update_module STATIC update_module/v3/update_module.cpp)
+add_library(update_module STATIC
+  update_module/v3/update_module.cpp
+  update_module/v3/update_module_download.cpp
+)
+target_link_libraries(update_module PUBLIC
+  common_log
+  common_conf
+  common_processes
+  mender_context
+  artifact
+)
 if(MENDER_USE_BOOST_FS)
   find_package(Boost 1.54 REQUIRED COMPONENTS filesystem)
   target_sources(update_module PRIVATE update_module/v3/platform/boost_filesystem/fs_operations.cpp)
   target_compile_options(update_module PRIVATE ${PLATFORM_SPECIFIC_COMPILE_OPTIONS})
   target_include_directories(update_module PRIVATE ${Boost_INCLUDEDIR})
-  target_link_libraries(update_module PUBLIC
-    common_log
-    common_conf
-    mender_context
-    artifact
-    ${Boost_LIBRARIES})
+  target_link_libraries(update_module PUBLIC ${Boost_LIBRARIES})
```

Comment:
Q: Why split htem, and keep both public, is it because `Boost` is external?

Reply:
It's because of the `if(MENDER_USE_BOOST_FS)`, I wanted to keep it inside the `if`, while our generic packages can go outside. I suppose you could argue that we don't really know if they are all needed when we build for a different platform. But if we keep it outside it's less likely that we forget to include something and break the build on another platform later. Either method has it's pros and cons, so I'm not really strongly arguing for either.

In any case, `PUBLIC` is always needed because it causes other targets that link to this one to include its dependencies as well, which is required when linking.